inc/Engine/Preload/Database/Queries/Cache.php

```diff
@@ -250,6 +250,22 @@ public function get_rows_by_url( string $url ) {
 		return $query;
 	}
 
+	/**
+	 * Get urls that match regex pattern.
+	 *
+	 * @param string $regex Regular Expression.
+	 * @return array
+	 */
+	public function get_urls_by_regex( string $regex ): array {
+		$db = $this->get_db();
+
+		$prefixed_table_name = $db->prefix . $this->table_name;
+		$query               = "SELECT `url` FROM `$prefixed_table_name` WHERE `status` = 'completed' AND `url` REGEXP '$regex'";
+		$results             = $db->get_results( $query, ARRAY_N );
+
```

Comment:
@engahmeds3ed Do you think it's the best way to do this query using BerlinDB? You have more experience with it than me